core/src/main/java/com/scalar/db/sql/metadata/CachedMetadata.java

```diff
@@ -0,0 +1,58 @@
+package com.scalar.db.sql.metadata;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.scalar.db.api.DistributedTransactionAdmin;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.sql.exception.SqlException;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public class CachedMetadata implements Metadata {
```

Comment:
I also extract metadata related interfaces in this PR.

The extracted interfaces are `Metadata`, `NamespaceMetadata`, `TableMetadata`, `ColumnMetadata`, and `IndexMetadata`.